runtime/compiler/env/J9CPU.cpp

```diff
@@ -24,22 +24,19 @@
 #pragma csect(STATIC,"J9J9CPU#S")
 #pragma csect(TEST,"J9J9CPU#T")
 
-#include "control/CompilationRuntime.hpp"
-#include "env/CompilerEnv.hpp"
 #include "env/CPU.hpp"
-#include "env/VMJ9.h"
-#include "j9.h"
-#include "j9port.h"
+#include "infra/Assert.hpp"                         // for TR_ASSERT
 
-OMRProcessorDesc
-J9::CPU::getProcessorDescription()
+const char *
+J9::CPU::getProcessorVendorId() 
    {
-#if defined(J9VM_OPT_JITSERVER)
-   if (auto stream = TR::CompilationInfo::getStream())
-      {
-      auto *vmInfo = TR::compInfoPT->getClientData()->getOrCacheVMInfo(stream);
-      return vmInfo->_processorDescription;
-      }
-#endif /* defined(J9VM_OPT_JITSERVER) */
-   return _processorDescription;
+   TR_ASSERT(false, "Vendor ID not defined for this platform!");
```

Comment:
Since this path is never supposed to be taken, you could transform this assert into a fatal one without the risk of adding extra overhead.